src/components/utils/DEDialog.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * A general purpose dialog
+ *
+ * @author sriram
+ *
+ */
+import React from "react";
+import { useTranslation } from "i18n";
+
+import { build } from "@cyverse-de/ui-lib";
+
+import ids from "./ids";
+
+import {
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    IconButton,
+    useMediaQuery,
+    useTheme,
+} from "@material-ui/core";
+
+import CloseIcon from "@material-ui/icons/Close";
+
+function DEDialog(props) {
+    const { baseId, open, title, onClose, content, actions } = props;
+    const { t } = useTranslation("common");
+    const theme = useTheme();
+    const fullScreen = useMediaQuery(theme.breakpoints.down("sm"));
+    const dialogTitleId = build(baseId, ids.DIALOG.TITLE);
+    return (
+        <Dialog
+            open={open}
+            onClose={onClose}
+            maxWidth="sm"
+            fullWidth
+            fullScreen={fullScreen}
+            aria-labelledby={dialogTitleId}
+        >
+            <DialogTitle>
+                {title}
+                <IconButton
+                    aria-label={t("close")}
+                    onClick={onClose}
+                    style={{
+                        position: "absolute",
+                        right: theme.spacing(0.5),
+                        top: theme.spacing(0.5),
+                        margin: 0,
+                    }}
+                >
+                    <CloseIcon />
+                </IconButton>
+            </DialogTitle>
+            <DialogContent>{content}</DialogContent>
```

Comment:
This works - I also personally wouldn't mind seeing the dialog content come from `children` instead of a `content` prop, but it's a small thing

Reply:
I would have gone with Children if it weren't for `dialogActions`. Should `dialogActions` be a `prop` or should that be Children too ? 🤔

Reply:
Interesting, react docs refer to the way I have it currently - https://reactjs.org/docs/composition-vs-inheritance.html

Reply:
I wouldn't include `actions` in the `children` too. Then it we would have to use it like this:
```jsx
<DEDialog
    baseId={...}
    open={...}
    title={...}
    ...
>
    <DialogContent>{...}</DialogContent>
    <DialogActions>{...}</DialogActions>
</DEDialog>
```
To my mind it makes sense to have the content as the `children`, and optional `actions` as a prop:
```jsx
<DEDialog
    baseId={...}
    open={...}
    title={...}
    actions={[...]}
    ...
>
    {...}
</DEDialog>
```